packages/flutter_tools/lib/src/compile.dart

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/globals.dart';
+import 'package:usage/uuid/uuid.dart';
+
+String _dartExecutable() {
+  final String engineDartSdkPath = artifacts.getArtifactPath(
+    Artifact.engineDartSdkPath
+  );
+  return fs.path.join(engineDartSdkPath, 'bin', 'dart');
+}
+
+Future<String> compile({String sdkRoot, String mainPath}) async {
+  final String frontendServer = artifacts.getArtifactPath(
+    Artifact.frontendServerSnapshotForEngineDartSdk
+  );
+
+  if (!sdkRoot.endsWith('/'))
```

Comment:
nit: use `{` and `}` (from [style guide][1]) here and other if-else below

[1]: https://www.dartlang.org/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures